### DISCUSSION TITLE
Problems in Copying GAMS model files for editing

### DISCUSSION THREAD

**Discussion by zsalimian** [iiasa/message_ix#416] [Getting started] [answered]
After Installing MESSAGEix from source, I decided to work with GAMS files containing the mathematical model core are installed with message_ix. therefore I used the following message-ix command-line program to copy the model files in a my own directory :
$ message-ix copy-model --set-default /path/for/model/files
As you see it copied just folders in the destination without any sub-folders and contents
How can I solve this problem?
![Untitled1](https://user-images.githubusercontent.com/71164512/97287319-aedff180-1859-11eb-8106-b24bd359bc17.png)

**Comment by francescolovat** [comment]
Hi @zsalimian ,

Where have you installed your model from source? In that folder you selected for installing MESSAGEix from source, can you find the files above in `/message_ix/model` ?

This command was designed to edit the GAMS files if you installed MESSAGEix through Anaconda/Miniconda. This way the GAMS files are quite hidden under a` site-packages` folder. But if you installed it from source, these files should be easy to find in the folder above and there is no need to use this command.

Please confirm that you can find those files before we continue.

**Comment by zsalimian** [comment]
Thank you.
No. I installed message-ix from source. The following address in the Github folder in my computer contains message_master gams file:
### \GitHub\main\message_ix#install-from-source-advanced-users\message_ix\model
But the problem is that I wanted to edit the gams file to build  simple model. As I mentioned above, I decided to create the same folder in another drive by using the following command.
$message-ix copy-model --set-default /path/for/model/files
but it didn't work..
Now, I copied the model folder manually in another drive and run message-master.gms. It requests GDX file. I do not know to tackle this problem.

**Comment by francescolovat** [comment]
Hi @zsalimian,

In regards of the `copy-model` faliure,
could you please specify which OS are you using? (Windows, Linux, etc.) and version?

In regards of the request GDX file:
Could you share the error log that requests the GDX in GAMS?
Have you read [this section of the documentation](https://docs.messageix.org/en/master/install.html?highlight=message_master.gms#from-source) in detail (step 8)?
These sections are designated to advanced users, we therefore assume that the user has the sufficient GAMS knowledge to be able to edit the `MESSAGE_master.gms` files to make it work outside Python (unfortunately we have no resources to teach MESSAGE user how to do so).

**Comment by zsalimian** [comment]
Hi,
Regarding copy-model faliure, my OS is windows 7.

Moreover, I am professional GAMS and Python user. Sorry, because I did not explain my problem exactly, you thought I am beginner  :))))
Firstly, I wanted to edit GAMS file itself, not working  via python. So I copied GAMS files manually in another folder in E drive. Then I run MESSAGE_master.gms. It shows this error:
## "Error 343 in E:\model\message/model_setup.gms" and the cursor showed  "$IF NOT EXIST '%in%' $ABORT "input GDX file '%in%' does not exist!" in model_setup.gms
I know that I should build a GDX file and put it in DATA folder.In GAMS projects,I usually provide an data file in excel format and use the following instructions to build GDX by model itself
$CALL GDXXRW.EXE example.xls par=excelsheet rng=excelsheet!A4:HT233 Rdim=2 Cdim=2
$GDXIN example.gdx
$LOAD excelsheet
$GDXIN
But I didn't find any excel data format in the folder to use it for entering data and building GDX. I thought I should build it myself. But I guess that the MESSAGE_master doesn't use excel to build GDX itself.

Secondly, after failing in the above process. I decided to work with it in python. I opened core file to run and edit with PyCharm. I am not sure whether I must edit core python file or not but it gives me the following error:
 Error running '__init__ (1)': Cannot run program "C:\Users\zsalimian\AppData\Local\Programs\Python\Python38\python.exe" (in directory "E:\Users\zsalimian\Documents\GitHub\main\message_ix#install-from-source-advanced-users\message_ix"): CreateProcess error=2, The system cannot find the file specified

Regards

**Comment by khaeru** [comment, marked as answer]
> I wanted to edit GAMS file itself, not working via python.

- If you want to run a manually-edited copy of the GAMS files, [the install instructions explain](https://docs.messageix.org/en/stable/install.html#copy-gams-model-files-for-editing) that you can do this **through** the message_ix Python package, by configuring it correctly.
- If you really need to run via the command line, you should not be using MESSAGE_master.gms but MESSAGE_run.gms; that file contains instructions for its own use.
- However, you **still** must use the Python API to generate a complete and compatible GDX file.

> I usually provide an data file in excel format and use the following instructions to build GDX by model itself.

message_ix uses the related ixmp package and the GAMS APIs to automatically generate input files and read output files in GDX format. Constructing the file using GDXXRW is complex and error-prone; it is not an intended use-case, and not supported.

> I am not sure whether I must edit core python file

No, this is not the way to use the software.

While manually editing the GAMS code is possible, it is almost never necessary. I would suggest reading the documentation thoroughly, from its beginning. This would include running through at least one of the tutorials. That will clarify the intended and supported ways of using the software, which cover almost all use-cases.
